### PR TITLE
Adds Spam Filter to PDA Message Server

### LINKED ISSUE #146: AI receives excessive PDA spam
**Issue Description**:
When the random "PDA spam" event occurs, the AI receives an excessive number of "intercepted" spam messages -- seemingly far more than an ordinary PDA-bearing player would.

**What did you expect to happen**: 
Only as much PDA spam as an ordinary player receives, or some action that can be taken to resolve it. I've heard that other players can get a new PDA to rid themselves of spam, but the AI does not have that option.

**What happened instead**:
The AI receives a ton of spam, and it seems to scale with the number of players. There's no way to stop it short of turning off the PDA messaging server, which affects all other players. I have occasionally had to do this claiming Law 3 because it was simply too distracting.

**Why is this bad/What are the consequences**:
The AI already has to sift through a huge amount of communication and split their attention between multiple simultaneous conversations. Forcing AI players to also sort through the PDA spam makes the role even harder, but it's not an interesting challenge, merely an annoying one. This is especially bad during times of high action, when everyone wants the AI's attention and assistance. It makes it more likely that the AI player will miss real players' requests, disrupting the fiction that they are a machine intelligence capable of promptly handling multiple issues at once.

**Steps to reproduce the problem**:
1. Play as AI.
2. Wait for the "PDA spam" event to occur or trigger it manually with admin powers. It appears to be very common, perhaps 50% of rounds I've played as AI.
3. Slowly go insane.

**When did the problem start happening**:
As long as I've been playing AI on this server.

**Extra information**:
Ideally, the AI should have some action they can take (such as accessing the PDA messaging server and clicking a button to "reset encryption" or something). If this also solved the event for the other players, it would change it from pure annoyance into an opportunity for the AI (or research director) to help out the entire station.

Alternatively, I recommend removing the PDA spam event from the AI, and maybe even for all players entirely. Other than giggling about wetskrell.rx, I can't see how it's a fun event to deal with for anyone, AI or otherwise.

### PR DESCRIPTION
**Pull request by blinkdog**
## What Does This PR Do
tl;dr: PDA Spam can be countered by turning the PDA Message Server off and on again.

* Adds a Spam Filter flag to the PDA Message Server
* When the PDA Spam event starts, it can send spam to the PDA Message Server
* Turning the PDA Message Server off and on again causes the spam filter to activate.
* If the spam filter is activated, the event will stop sending messages to the message server, ending PDA spam.

## Why It's Good For The Game
Both the AI and RD can do something about the PDA Spam event. (This Fixes #146)

## Changelog
:cl:
add: Power cycling PDA Message Server activates the Spam Filter.
/:cl:

### CODE REVIEW

**Comment by blinkdog** [comment]
Ack. I just realized two things:

1. You need to be quick on the reboot to get a useful filter upgrade

If the PDA Spam event sees the server turned off, it will attempt to pick another one. If this happens while the Message Server is off, it will choose the Message Server again and reset the spam level. If you don't turn the Message Server back on quickly enough, the filter upgrade won't help.

2. A solution to PDA Spam already exists

Just turn the Message Server off for five minutes. After the event fails to spam anyone for five minutes, the event automatically ends. So the event really is a choice between the crew being without PDA messages for five minutes, or keeping PDA messaging but suffering through PDA spam for the next hour.

**Comment by ClockworkGolem** [comment]
Fantastic! Thanks!

**Comment by ZomgPonies** [comment]
@blinkdog Is this ready for merge?

**Comment by blinkdog** [comment]
Not quite yet; I'd like to fix up that quick reboot problem before this gets merged.
